rpc/http.go

```diff
@@ -40,8 +40,14 @@ const (
 
 var nullAddr, _ = net.ResolveTCPAddr("tcp", "127.0.0.1:0")
 
+// HttpClient contains the methods of http.Client that we require.
+// This allows users to provide wrapped HTTP clients for use with the RPC.
+type HttpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
```

Comment:
I don't think it's a good idea to define this interface here, as any eventual extra method (that's provided by Go's http.Client) will result in breaking the interface. Let's just directly require specifying a `*http.Client`. That way we're future proof.

Reply:
Yeah had that first, but it makes it impossible to use wrappers of an http.Client like proxies or monitors.

Reply:
AFAIK that's what [`http.Client.Transport`](https://golang.org/pkg/net/http/#Client) is for:

        // Transport specifies the mechanism by which individual
        // HTTP requests are made.
        // If nil, DefaultTransport is used.
        Transport RoundTripper

You can define all proxy and monitoring methods inside the http Client and that's how other libraries use it too (e.g. Google's Go APIs https://github.com/google/google-api-go-client/issues/146#issuecomment-228140570).

Reply:
Hmm, didn't know about the Transport field. That's a smart thing they did there. Will change, one sec.